### PR TITLE
Add ability to pass in a11yTitle in dropProps for Menu

### LINKED ISSUE #5964: Menu accessibility
Menus should be accessible.  In general, all menus should have proper roles, but I think it makes sense to enable a menu to accept an `a11yTitle` that is the name of the menu, using heuristics to name the aspects of the menu (e.g. "Open Primary Menu" on the button when closed should become "Primary Menu" on the menu and "Close Primary Menu" on the button when open in case the menu is a persistent one)

### Expected Behavior

Menus should have a role of menu by default and either intelligently apply aria attributes or allow greater out of the box support for them. TypeScript also shouldn't error when you override any of these values.

### Actual Behavior

I was working on writing a Cypress test and discovered that my search for an element with `role=menu` was failing when trying to target the opened menu. In attempting to set the role attribute on the menu, I discovered that TypeScript errors when you attempt to pass HTML attributes in the `dropProps` prop despite applying them correctly. 

### URL, screen shot, or Codepen exhibiting the issue

image

image

image

If a repro is valuable, I can add it, but we're sprinting toward a release and I don't really have a ton of time to spare right now.

### Steps to Reproduce

1. Use a `Menu` in a TypeScript environment
2. Inspect the element and note the lack of a `role`
3. Try to apply the role and cry because TypeScript pulls out the red underline of fury

### Your Environment

- Grommet version: 2.20.1
- Browser Name and version: Chrome (latest)
- Operating System and version (desktop or mobile): Desktop MacOS

### PR DESCRIPTION
**Pull request by amandadupell**
#### What does this PR do?

Adds an optional a11yTitle prop to DropProps to allow users to specify an a11yTitle that way

#### Where should the reviewer start?

`Menu.js`

#### Do Jest tests follow these best practices?

- [ ] `screen` is used for querying.
- [ ] The correct query is used. (Refer to [this list of queries](https://testing-library.com/docs/queries/about/#priority))
- [ ] `userEvent` is used in place of `fireEvent`.
- [ ] `asFragment()` is used for snapshot testing.

#### What are the relevant issues?

closes #5964 

#### Do the grommet docs need to be updated?

No

#### Should this PR be mentioned in the release notes?

Yes

#### Is this change backwards compatible or is it a breaking change?

Compatible

### CODE REVIEW

**Review by jcfilben — COMMENTED** [review]
What if we have a case where Menu is given an `a11yTitle` and an `a11yTitle` is passed through drop props? Currently it looks like the dropProps `a11yTitle` will override the menu `a11yTitle`

**Review by jcfilben — APPROVED** [review]
Looks good!

File: src/js/components/Drop/index.d.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {
 } from '../../utils';
 
 export interface DropProps {
+  a11yTitle?: string;
```
Comment (britt6612) [inline src/js/components/Drop/index.d.ts:11]: can we use the a11yTitle type here ?

File: src/js/components/Menu/stories/typescript/Children.tsx [hunk 2]
```diff
@@ -10,9 +10,12 @@ const MenuWithChildren = (props) => (
       { label: 'Launch', onClick: () => {} },
       { label: 'Abort', onClick: () => {} },
     ]}
+    dropProps={{
+      a11yTitle: 'Test',
```
Comment (britt6612) [inline src/js/components/Menu/stories/typescript/Children.tsx:14]: if we are going to keep this story can we add something else other than `Test` or remove

**Review by britt6612 — APPROVED** [review]

File: src/js/components/Menu/Menu.js [hunk 3]
```diff
@@ -332,11 +330,7 @@ const Menu = forwardRef((props, ref) => {
               {alignControlMirror === 'top' && align.top === 'top'
                 ? controlMirror
                 : undefined}
-              <Box
-                overflow="auto"
-                role="menu"
-                a11yTitle={ariaLabel || a11yTitle}
-              >
+              <Box overflow="auto" role="menu" a11yTitle={dropA11y || a11y}>
```
Comment (ericsoderberghp) [inline src/js/components/Menu/Menu.js:333]: We are already passing `dropProps` to the DropButton at line 312. I think anything passed via `dropProps` should be handled by Drop itself. In this particular case, I think we should be changing DropContainer.js to set the `aria-label` on StyledDrop around line 331.

**Pull request merged**